pkg/kbuser/space_ids.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kbuser
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	informerscorev1 "k8s.io/client-go/informers/core/v1"
+	kubernetes "k8s.io/client-go/kubernetes"
+	cache "k8s.io/client-go/tools/cache"
+)
+
+const KubeBindLabelName string = "kubestellar.io/kube-bind-id"
+
+const CMNamePrefix string = "kbmap-"
+
+// KubeBindSpaceRelation is specific to one kube-bind service provider
+// and reveals the 1:1 relation between the kube-bind identifier for
+// a consumer (i.e., the name of the "cluster namespace" for that consumer
+// in the provider) and the underlying space identifier (i.e., the
+// `logicalcluster.Name`).
+type KubeBindSpaceRelation interface {
+	// SpaceIDToKubeBind maps an underlying space ID to a kube-bind space ID.
+	// Returns empty string if there is no relationship.
+	SpaceIDToKubeBind(spaceID string) string
+
+	// SpaceIDFromKubeBind maps a kube-bind space ID to an underlying space ID.
+	// Returns empty string if there is no relationship.
+	SpaceIDFromKubeBind(kubeBindID string) string
+}
+
+// NewKubeBindSpaceRelation creates a KubeBindSpaceRelation given a
+// Kubernetes client for the provider's space and a context bounding
+// the relation's lifetime.
+func NewKubeBindSpaceRelation(ctx context.Context, client kubernetes.Interface) *kubeBindSpaceRelation {
+	informer := informerscorev1.NewFilteredConfigMapInformer(client, "kubestellar-system", 0, cache.Indexers{}, tweakListOptions)
```

Comment:
Maybe we can continue to use `kubestellar` namespace?

Reply:
yes